sympy/simplify/simplify.py

```diff
@@ -1101,6 +1106,59 @@ def bottom_up(rv, F, atoms=False, nonbasic=False):
     return rv
 
 
+def kroneckersimp(expr):
+    """
+    Simplify expressions with KroneckerDelta.
+
+    The only simplification currently attempted is to identify multiplicative cancellation:
+
+    >>> from sympy import KroneckerDelta, kroneckersimp
+    >>> from sympy.abc import i, j
+    >>> kroneckersimp(1 + KroneckerDelta(0, j) * KroneckerDelta(1, j))
+    1
+    """
+    def args_cancel(args1, args2):
+        for i1 in range(2):
+            for i2 in range(2):
+                a1 = args1[i1]
+                a2 = args2[i2]
+                a3 = args1[(i1 + 1) % 2]
+                a4 = args2[(i2 + 1) % 2]
+                if Eq(a1, a2) is S.true and Eq(a3, a4) is S.false:
+                    return True
+        return False
+
+    def cancel_kronecker_mul(m):
+        from sympy.utilities.iterables import subsets
+
+        args = m.args
+        deltas = [a for a in args if isinstance(a, KroneckerDelta)]
+        for delta1, delta2 in subsets(deltas, 2):
+            args1 = delta1.args
+            args2 = delta2.args
+            if args_cancel(args1, args2):
+                return 0*m
+        return m
+
+    def is_mul_with_kronecker(e):
+        return isinstance(e, Mul) and e.has(KroneckerDelta)
```

Comment:
Do you mean `...and any(isinstance(a, KroneckerDelta) for a in e.args)`? But I'm not sure that doing this test for KroneckerDelta twice is worth the trouble: if it is a Mul then the `cancel_kronecker_mul` will be called and will gracefully pass if `deltas` is `[]`:
```python
>>> list(subsets([],2))
[]
```